### ISSUE TITLE
Receiving NoResponse Errors when running commands that take a longer time

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#32305]
### Description of Issue/Question
Receiving a no response error when running commands that take longer then the default 5 second timeout. 

### Steps to Reproduce Issue
1. Verify you can communicate with the minion using a simple test.ping

```
➜  salt git:(9c1b2a4) sudo salt '*ch3ll-mast*' test.ping    
ch3ll-master:
    True
```

2. Then run a command that takes longer then the default 5 second timeout to run you receive the error:

```
➜  salt git:(tagv2015.8.8) sudo salt '*ch3ll-mast*' test.sleep 30
ch3ll-master:
    Minion did not return. [No response]
```

### Versions Report

```
➜  salt git:(9c1b2a4) salt --versions-report
Salt Version:
           Salt: 2016.3.0rc1-1796-g9c1b2a4
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.3.1
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 3.2.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.4.5.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

Also here is the git bisect:

```
➜  salt git:(5d970ca) git bisect bad
5d970ca031687eefe8161997728a5cf54a3dc781 is the first bad commit
commit 5d970ca031687eefe8161997728a5cf54a3dc781
Author: charles_paclat <charles.paclat@oracle.com>
Date:   Fri Mar 25 11:40:24 2016 -0400

    fixes 29817
    
    Makes sure we unsubscribe from self.events and also dereference the job info iterator

:040000 040000 be787578819068c5cd7048c19e1c8c724c989f6e 44089fb6998ba0fee30681d559d81b9dc4345aea M      salt
```

My good commit was :e648464f33a3821c24a6ad50c93fbee874a7953f
bad commit :9c1b2a43a9f746416b3ab49d5fcbf7935064c850

I also tested on the head of the 2016.3 branch today and its still not workng:

```
➜  salt git:(branch2016.3) salt --version
salt 2016.3.0rc1-1803-g7ae89b0 (Boron)
```

**Comment by meggiebot** [comment]
@cachedout if you get a chance to look at this go ahead, @rallytime is going to take the first swing over the weekend

**Comment by Ch3LL** [comment]
This is also broken at the head of 2015.8

```
➜  salt git:(branch2015.8) sudo salt '*ch3ll-mast*' test.sleep 30         
ch3ll-master:
    Minion did not return. [No response]
➜  salt git:(branch2015.8) salt --version
salt 2015.8.8-205-g664043d (Beryllium)
➜  salt git:(branch2015.8) 
```

**Comment by meggiebot** [comment]
Per Nicole this is at the head of 2015.8, but not in the packaged release 2015.8.8

**Comment by rallytime** [comment]
@Ch3LL #32332 should fix this issue.

**Comment by Ch3LL** [comment]
I verified that this is working by testing your PR:

```
➜  salt git:(test-32332) sudo salt '*ch3ll-mast*' test.ping    
ch3ll-master:
    True
➜  salt git:(test-32332) sudo salt '*ch3ll-mast*' test.sleep 30
ch3ll-master:
    True
➜  salt git:(test-32332) salt --version
salt 2015.8.8-204-g8dc1161 (Beryllium)
```

I would like to keep this open so I can test this once its merged into 2015.8 and merged forward to 2016.3.

**Comment by rallytime** [comment]
Oh, I pulled the fix into 2016.3 with #32337. So once that is merged in, you should be able to test off of the 2016.3 branch, too.

**Comment by Ch3LL** [comment]
Awesome! I'll test both of them when they are merged :)

**Comment by Ch3LL** [comment]
I just tested on both the head of 2015.8 and 2016.3 and this is now working. Thanks!
